extensions/reclaim-ai/src/notifications.tsx

```diff
@@ -91,29 +96,28 @@ export default function Command() {
 
   const eventData = filterMultipleOutDuplicateEvents(eventsResponse);
 
-  const showDeclinedEvents = useMemo(() => {
-    return !!currentUser?.settings.showDeclinedEvents;
-  }, [currentUser]);
-
-  const eventMoment: ApiResponseMoment = useMemo(() => {
-    const now = new Date();
-    const events = eventData
-      ?.filter((event) => {
-        return showDeclinedEvents ? true : event.rsvpStatus !== "Declined" && event.rsvpStatus !== "NotResponded";
-      })
-      .filter((event) => {
-        return event.reclaimEventType !== "CONF_BUFFER" && event.reclaimEventType !== "TRAVEL_BUFFER";
-      })
-      .filter((event) => isAfter(new Date(event.eventEnd), now))
-      .filter((event) => {
-        return !(differenceInHours(new Date(event.eventEnd), new Date(event.eventStart)) >= 24);
-      });
+  const { data: eventMomentData, isLoading: isLoadingMoment } = useFetch<ApiResponseMoment>(`${apiUrl}/moment/next`, {
+    headers: fetchHeaders,
+    keepPreviousData: true,
+  });
 
+  // if the events returned my moment/next are synced events then return the original event from the events call if it exists
+  const eventMoment = useMemo(() => {
+    if (!eventMomentData) return eventMomentData;
+    const { event, nextEvent } = eventMomentData;
     return {
-      event: events?.at(0),
-      nextEvent: events?.at(1),
+      event: event
+        ? eventData?.find((event) => event.eventId === getOriginalEventIDFromSyncEvent(event)) ?? event
+        : null,
+      nextEvent: nextEvent
+        ? eventData?.find((event) => event.eventId === getOriginalEventIDFromSyncEvent(nextEvent)) ?? nextEvent
+        : null,
```

Comment:
hey @pernielsentikaer i am unsure how the background sync works in reclaim, do you have any insights how many react render cycles are happening when updating the menubar. The reason i am asking i think if the current setup here, it could be the case that first `useFetch` has no data and the second one does which therefore would show an incorrect result in the menubar. What would be the best practice to exclude this state?

Reply:
@shkreios While `isLoading` is set to `true`, you have as many renders as needed. Once it's set to `false`, you are only guaranteed one more render 🙂